### DISCUSSION TITLE
Return pdf response.

### DISCUSSION THREAD

**Discussion by mmarvb8h** [jeremyevans/roda#312] [General]
How would i return a pdf file? Do i just set content header to application/pdf? 

Is there a plugin for this?

thx ~Marvin
email: mmarvb7@gmail.com

**Comment by cf-ka** [comment]
You can set the content Header by your self.
Maybe you can use the Sinatra_helpers from the plugin list, this contains a send_file method and is a solution for that you looking for.
Link to the Docs - Sinatra_Helpers : http://roda.jeremyevans.net/rdoc/classes/Roda/RodaPlugins/SinatraHelpers.html#module-Roda::RodaPlugins::SinatraHelpers-label-send_file

greetings _ka_

**Comment by jeremyevans** [comment]
`response['Content-Type'] = 'application/pdf'` should be sufficient if your route block is returning the PDF data as a string.

**Reply by mmarvb8h** [reply]
thx for ur response. here is an example of what i'm doing in Sinatra:

    unless new_pdf.nil?
      byebug
      stream(:keep_open) do |out|
        out << new_pdf
      end
    end

I think in a Rails app i've used:
send_data(your_generated_pdf, ... )

So i guess i'm looking for something like that.

**Reply by jeremyevans** [reply]
Roda has a `streaming` plugin you can use for that (https://roda.jeremyevans.net/rdoc/classes/Roda/RodaPlugins/Streaming.html).  However, if you are not generating your PDF in chunks, you are better off just having your route block return `new_pdf`.

**Comment by mmarvb8h** [comment]
Setting the header and just returning new_pdf seems to work fine... thx...
